### PR TITLE
[Refact](inverted index) refact inverted index file writer to make it more clear

### PR DESCRIPTION
**Pull request by airborne12**
## Proposed changes

Issue Number: close #xxx

1. This PR refactors path handling and logging in the InvertedIndexDescriptor and related components, ensuring clarity and consistency across the codebase.
2. Updated initialization logic for DorisFSDirectory and DorisRAMFSDirectory, removing confusing cfs and fs logic.
3. Removed DorisCompoundFileWriter and consolidated its logic into InvertedIndexFileWriter.
3. Removed unused methods get_inverted_index_size and file_size from several classes.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by airborne12** [comment]
run buildall

**Comment by airborne12** [comment]
run buildall

**Review by qidaye — APPROVED** [review]
LGTM

**Review by qidaye — DISMISSED** [review]
LGTM

File: be/src/olap/rowset/segment_v2/inverted_index_fs_directory.cpp [hunk 1]
```diff
@@ -341,36 +340,22 @@ DorisFSDirectory::DorisFSDirectory() {
...
+    directory = path;
     if (lock_factory == nullptr) {
         lock_factory = _CLNEW lucene::store::NoLockFactory();
     }
 
     lucene::store::Directory::setLockFactory(lock_factory);
 
     // It's fail checking directory existence in S3.
-    if (fs->type() == io::FileSystemType::S3) {
+    if (_fs->type() == io::FileSystemType::S3) {
```
Comment (csun5285) [inline be/src/olap/rowset/segment_v2/inverted_index_fs_directory.cpp:354]: S3 file system now supports the `exists`

File: be/src/olap/rowset/segment_v2/inverted_index_fs_directory.cpp [hunk 2]
```diff
@@ -82,7 +82,6 @@
 namespace doris::segment_v2 {
 
 const char* const DorisFSDirectory::WRITE_LOCK_FILE = "write.lock";
-const char* const DorisFSDirectory::COMPOUND_FILE_EXTENSION = ".idx";
 
 class DorisFSDirectory::FSIndexOutput : public lucene::store::BufferedIndexOutput {
```
Comment (csun5285) [inline be/src/olap/rowset/segment_v2/inverted_index_fs_directory.cpp:86]: `DorisFSDirectory::FSIndexInput` is declared in `inverted_index_fs_directory.h`, while `DorisFSDirectory::FSIndexOutput` is declared in `inverted_index_fs_directory.cpp`.

File: be/src/olap/task/index_builder.cpp [hunk 3]
```diff
@@ -251,7 +251,7 @@ Status IndexBuilder::handle_single_rowset(RowsetMetaSharedPtr output_rowset_meta
         const auto& output_rs_tablet_schema = output_rowset_meta->tablet_schema();
         if (output_rs_tablet_schema->get_inverted_index_storage_format() !=
             InvertedIndexStorageFormatPB::V1) {
-            const auto& fs = io::global_local_filesystem();
+            const auto& fs = output_rowset_meta->fs();
```
[the remark this answers is not in the document]
Comment (airborne12) [reply be/src/olap/task/index_builder.cpp:254]: Currently, this will not cause any bugs because we do not support index building in cloud mode.

**Review by zzzxl1993 — APPROVED** [review]
LGTM

File: be/src/olap/rowset/segment_v2/inverted_index_writer.cpp [hunk 4]
```diff
@@ -537,8 +537,6 @@ class InvertedIndexColumnWriterImpl : public InvertedIndexColumnWriter {
         return 0;
     }
 
-    int64_t file_size() const override { return _dir->getCompoundFileSize(); }
```
[the remark this answers is not in the document]
Comment (airborne12) [reply be/src/olap/rowset/segment_v2/inverted_index_writer.cpp:540]: no use in future

File: be/src/olap/task/index_builder.cpp [hunk 5]
```diff
@@ -283,12 +283,10 @@ Status IndexBuilder::handle_single_rowset(RowsetMetaSharedPtr output_rowset_meta
                                                      std::move(inverted_index_file_writer));
             }
             for (auto&& [seg_id, inverted_index_writer] : _inverted_index_file_writers) {
-                LOG(INFO) << "close inverted index file "
-                          << inverted_index_writer->get_index_file_path();
                 auto st = inverted_index_writer->close();
                 if (!st.ok()) {
-                    LOG(ERROR) << "close inverted index file "
-                               << inverted_index_writer->get_index_file_path() << " error:" << st;
+                    LOG(ERROR) << "close inverted_index_writer error:" << st;
```
[the remark this answers is not in the document]
Comment (airborne12) [reply be/src/olap/task/index_builder.cpp:288]: file path info is in error message itself.

**Comment by airborne12** [comment]
run buildall

**Comment by airborne12** [comment]
run buildall

**Review by qidaye — APPROVED** [review]

**Pull request merged**